sdk/inc/azure/iot/az_iot_pnp_client.h

```diff
@@ -0,0 +1,670 @@
+    size_t mqtt_user_name_size,
+    size_t* out_mqtt_user_name_length)
+{
+  return az_iot_hub_client_get_user_name(
+      &(client->_internal.iot_hub_client),
+      mqtt_user_name,
+      mqtt_user_name_size,
+      out_mqtt_user_name_length);
+}
+
+/**
+ * @brief Gets the MQTT client id.
+ *
+ * The client id will be of the following format:
+ *
+ * `{device_id}`
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[out] mqtt_client_id A buffer with sufficient capacity to hold the MQTT client id.
+ *                            If successful, contains a null-terminated string with the client id
+ *                            that needs to be passed to the MQTT client.
+ * @param[in] mqtt_client_id_size The size, in bytes of \p mqtt_client_id.
+ * @param[out] out_mqtt_client_id_length __[nullable]__ Contains the string length, in bytes, of
+ *                                                      of \p mqtt_client_id. Can be `NULL`.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p mqtt_client_id must not be `NULL`.
+ * @pre \p mqtt_client_id_size must be greater than 0.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_get_client_id(
+    az_iot_pnp_client const* client,
+    char* mqtt_client_id,
+    size_t mqtt_client_id_size,
+    size_t* out_mqtt_client_id_length)
+{
+  return az_iot_hub_client_get_client_id(
+      &client->_internal.iot_hub_client,
+      mqtt_client_id,
+      mqtt_client_id_size,
+      out_mqtt_client_id_length);
+}
+
+/*
+ *
+ * SAS Token APIs
+ *
+ *   Use the following APIs when the Shared Access Key is available to the application or stored
+ *   within a Hardware Security Module. The APIs are not necessary if X509 Client Certificate
+ *   Authentication is used.
+ */
+
+/**
+ * @brief Gets the Shared Access clear-text signature.
+ * @details The application must obtain a valid clear-text signature using this API, sign it using
+ *          HMAC-SHA256 using the Shared Access Key as password then Base64 encode the result.
+ *
+ * @param[in] client The #az_iot_pnp_client to use for this call.
+ * @param[in] token_expiration_epoch_time The time, in seconds, from 1/1/1970.
+ * @param[in] signature An empty #az_span with sufficient capacity to hold the SAS signature.
+ * @param[out] out_signature The output #az_span containing the SAS signature.
+ *
+ * @pre \p client must not be `NULL`.
+ * @pre \p token_expiration_epoch_time must be greater than 0.
+ * @pre \p signature must be a valid, non-empty #az_span.
+ * @pre \p out_signature must not be `NULL`. It must point to a valid #az_span instance.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ */
+AZ_NODISCARD AZ_INLINE az_result az_iot_pnp_client_get_sas_signature(
+    az_iot_pnp_client const* client,
+    uint32_t token_expiration_epoch_time,
+    az_span signature,
+    az_span* out_signature)
```

Comment:
>out_signature [](start = 13, length = 13)

this is the document that needs to be HMAC?

Reply:
yes